File: todoapp.client/src/types/StickyNote.ts
import { User } from "./User";

export interface IStickyNote {
    id: number;
    userId: number;
    name?: string;
    details?: string;
    user?: User;
}
export class StickyNote implements IStickyNote {
    constructor(
        public id: number,
        public userId: number,
        public name?: string,
        public details?: string,
        public user?: User,
    ) {
    }
}

File: todoapp.server/Models/StickyNote.cs
using System;
using System.Collections.Generic;

namespace todoapp.server.Models;

public partial class StickyNote
{
    public int Id { get; set; }

    public int UserId { get; set; }

    public string? Name { get; set; }

    public string? Details { get; set; }

    public virtual User User { get; set; } = null!;
}
